query-planner-js/src/buildPlan.ts

```diff
@@ -3059,6 +3080,7 @@ export class QueryPlanner {
       statistics,
       inconsistentAbstractTypesRuntimes: this.inconsistentAbstractTypesRuntimes,
       config: this.config,
+      overrideConditions: options?.overrideConditions ?? new Map(),
```

Comment:
One caveat to note here is that if the user forgets to provide a label, then certain fields will be impossible to traverse (and query planning will be fail). It may be good to add code here to manually set such labels to be `false` if not present. (You'll want to precompute all the labels in a supergraph schema inside the constructor for `QueryPlanner`, by looking at the applications of the `@join__field directive.)

Reply:
f7646134